### PR TITLE
Add improvements to Luigi Client API comments and annotations

### PR DESCRIPTION
**Pull request by bszwarc**
**Description**

Changes proposed in this pull request:

- Add improvements to Luigi API docs 

**Related issues**

See #170

### CODE REVIEW

File: client/luigi-client.js [hunk 1]
```diff
@@ -169,8 +169,8 @@
       return id;
     },
     /**
-     * Removes a init listener
-     * @param {string} id the id that was returned by `addInitListener`
+     * Removes the init listener.
```
Comment (maxmarkus) [inline client/luigi-client.js]: There can be multiple init listeners, the function removes the listener based on the id. 
The @param change is fine, `the` is ok here.

File: client/luigi-client.js [hunk 2]
```diff
@@ -196,8 +196,8 @@
       return id;
     },
     /**
-     * Removes a context update listener
-     * @param {string} id the id that was returned by `addContextUpdateListener`
+     * Removes the context update listener.
```
Comment (maxmarkus) [inline client/luigi-client.js]: Same as above, it removes a specific listener, based on its id.

File: client/luigi-client.js [hunk 3]
```diff
@@ -310,7 +310,7 @@
         },
 
         /**
-         * Sets the current navigation context, which is then used by the `navigate` function. This has to be a parent navigation context, it is not possible to go to child navigation contexts.
+         * Sets the current navigation context which is then used by the `navigate` function. This has to be a parent navigation context, it is not possible use the child navigation contexts.
```
Comment (dariadomagala) [inline client/luigi-client.js]: Shouldn't that be
```suggestion
         * Sets the current navigation context which is then used by the `navigate` function. This has to be a parent navigation context, it is not possible to use the child navigation contexts.
```

File: client/luigi-client.js (hunk 1 above)
Comment (bszwarc) [reply client/luigi-client.js]: Thank you, updated.

File: client/luigi-client.js (hunk 3 above)
Comment (bszwarc) [reply client/luigi-client.js]: Yes, you are right - updated this one.

File: client/luigi-client.js (hunk 2 above)
Comment (bszwarc) [reply client/luigi-client.js]: Done.

File: client/luigi-client.js [hunk 4]
```diff
@@ -169,8 +169,8 @@
       return id;
     },
     /**
-     * Removes a init listener
-     * @param {string} id the id that was returned by `addInitListener`
+     * Removes an init listener.
+     * @param {string} id the id that was returned by the `addInitListener` function
```
Comment (dariadomagala) [inline client/luigi-client.js]: `id` or `ID`? There is an inconsistency compared to the line 200

Comment (bszwarc) [reply client/luigi-client.js]: let's stay with `id`.

**Comment by bszwarc** [comment]
Fixes #170

**Pull request merged**